libsolidity/analysis/TypeChecker.cpp

```diff
@@ -453,7 +453,12 @@ bool TypeChecker::visit(VariableDeclaration const& _variable)
 		if (contractType->contractDefinition().isLibrary())
 			m_errorReporter.typeError(1273_error, _variable.location(), "The type of a variable cannot be a library.");
 	if (_variable.value())
-		expectType(*_variable.value(), *varType);
+	{
+		if (_variable.isStateVariable() && dynamic_cast<MappingType const*>(varType))
+			m_errorReporter.typeError(6280_error, _variable.location(), "Mappings cannot be assigned to.");
```

Comment:
`expectType` descends into `*_variable.value()`.  I think we should do that manually here, otherwise we get internal errors.